serde_json_path_core/src/spec/integer.rs

```diff
@@ -52,32 +52,30 @@ impl Integer {
         Self::try_new(value).expect("value is out of the valid range")
     }
 
-    /// Take the absolute value, producing `None` if the resulting value is outside
-    /// the valid range [-(2<sup>53</sup>)+1, (2<sup>53</sup>)-1].
-    pub fn checked_abs(mut self) -> Option<Self> {
-        self.0 = self.0.checked_abs()?;
-        self.check().then_some(self)
+    /// Take the absolute value, producing a new instance of [`Integer`]
+    pub fn abs(&self) -> Self {
+        Self(self.0.abs())
```

Comment:
Might be useful to have a comment here explicitly mentioning that this can never fail? For example something like:
```rust
// Can never fail because Integer `MIN.abs() == MAX` (and MIN > i64::MIN)
```

But might not be needed, maybe this is clear enough already.